### PR TITLE
added /api/bot/online API

### PR DESCRIPTION
**Pull request by kalpgarg**
Added REST endpoint for /api/bot/online.

This is to implement the first task in the issue https://github.com/lichess-org/berserk/issues/6. 

This is my first PR for this project. If this PR is ok, I plan to implement all the tasks under this issue.

All tests are getting passed when run locally.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -125,6 +125,7 @@ Most of the API is available:
     client.board.claim_victory
     client.board.go_berserk
 
+    client.bots.online
```
Comment (icp1994) [inline README.rst:128]:
```suggestion
    client.bots.get_online_bots
```

File: berserk/clients.py [hunk 2]
```diff
@@ -1077,6 +1077,17 @@ def stream_game_state(self, game_id: str) -> Iterator[Dict[str, Any]]:
         path = f"api/bot/game/stream/{game_id}"
         yield from self._r.get(path, stream=True, converter=models.GameState.convert)
 
+    def get_online_bots(self, limit: int | None = None) -> Iterator[Dict[str, Any]]:
+        """
+        Stream the online bot users.
+
+        :param limit: Maximum number of bot users to fetch
```
Comment (icp1994) [inline berserk/clients.py:1084]:
```suggestion
        :param limit: Maximum number of bot users to fetch
        :type limit: int, optional
```

File: berserk/clients.py [hunk 3]
```diff
@@ -1077,6 +1077,17 @@ def stream_game_state(self, game_id: str) -> Iterator[Dict[str, Any]]:
...
+        :return: Iterator over the results
```
Comment (icp1994) [inline berserk/clients.py:1085]:
```suggestion
        :return: Iterator over the results
        :return: iterator over online bots
```

File: berserk/clients.py [hunk 4]
```diff
@@ -1077,6 +1077,17 @@ def stream_game_state(self, game_id: str) -> Iterator[Dict[str, Any]]:
...
+        """
+        path = f"api/bot/online"
```
Comment (icp1994) [inline berserk/clients.py:1087]:
```suggestion
        path = "api/bot/online"
```

**Review by icp1994 — CHANGES_REQUESTED** [review]

**Comment by kalpgarg** [comment]
Thanks, @icp1994, for your suggestions. I incorporated them all in the last commit.

File: berserk/clients.py [hunk 5]
```diff
@@ -1077,6 +1077,21 @@ def stream_game_state(self, game_id: str) -> Iterator[Dict[str, Any]]:
         path = f"api/bot/game/stream/{game_id}"
         yield from self._r.get(path, stream=True, converter=models.GameState.convert)
 
+    def get_online_bots(self, limit: int | None = None) -> Iterator[Dict[str, Any]]:
+        """
+        Stream the online bot users.
+
+        :param limit: Maximum number of bot users to fetch
+        :type limit: int, optional
+        :return: Iterator over the results
```
Comment (icp1994) [inline berserk/clients.py:1086]:
```suggestion
```

**Review by icp1994 — COMMENTED** [review]
Thank you, looks good. There's just one line I forgot to remove in yesterday's review. After you are done, please squash/fixup your commits into one. The msg should be something like: "Add /api/bot/online endpoint"

**Comment by kalpgarg** [comment]
Done @icp1994

**Review by icp1994 — APPROVED** [review]

**Comment by benediktwerner** [comment]
Thanks! Release as v0.12.1

**Pull request merged**